components/sidebar/add_channel_dropdown.tsx

```diff
@@ -182,7 +157,27 @@ const AddChannelDropdown = ({
                 id='AddChannelDropdown'
                 ariaLabel={intl.formatMessage({id: 'sidebar_left.add_channel_dropdown.dropdownAriaLabel', defaultMessage: 'Add Channel Dropdown'})}
             >
-                {renderDropdownItems()}
+                <Menu.Group>
+                    {renderDropdownItems.map((item, index, items) => (
+                        <Menu.Group
+                            key={index}
+                            divider={<li className={item.divider}/>}
+                        >
+                            {item.isEligible &&
+                            <Menu.ItemAction
+                                id={item.id}
+                                onClick={item.onClick}
+                                icon={<i className={item.icon}/>}
+                                text={item.text}
+                                extraText={item.extraText}
+                                index={index}
+                                size={items.length}
```

Comment:
size holds both eligible and non eligible items, I think we should use filter before map to exclude non eligible and then size can be the `items.length`